### PR TITLE
updated: UI for over receiving(#85zrnfx68)

### LINKED ISSUE #148: UI for over receiving
## What is the motivation for adding/enhancing this feature?
When the user is receiving more than was ordered in a PO, or was expected in a Shipment, the progress bar should turn red/danger so that the user knows they are doing something unexpected.

### PR DESCRIPTION
**Pull request by disha1202**
### Related Issues

Closes #148 

### Screenshots of Visual Changes before/after (If There Are Any)

**IMPORTANT NOTICE** - Remember to add changelog entry

### Contribution and Currently Important Rules Acceptance

- [ ] I read and followed [contribution rules](https://github.com/hotwax/receiving#contribution-guideline)

### CODE REVIEW

File: src/views/PurchaseOrderDetail.vue [hunk 1]
```diff
@@ -72,7 +72,8 @@
             </div>
 
             <div class="qty-progress">
-              <ion-progress-bar :value="item.quantityAccepted/item.quantity" />
+              <!-- TODO: improve the handling of quantityAccepted -->
+              <ion-progress-bar :color="parseInt(item.quantityAccepted)/(item.quantity - getPOItemAccepted(item.productId)) > 1 ? 'danger' : 'primary'" :value="(parseInt(item.quantityAccepted) +  getPOItemAccepted(item.productId))/(item.quantity)" />
```
Comment (ymaheshwari1) [inline src/views/PurchaseOrderDetail.vue:76]: We can create a computed property instead of defining logic in template.

**Review by ymaheshwari1 — CHANGES_REQUESTED** [review]

File: src/views/PurchaseOrderDetail.vue (hunk 1 above)
Comment (disha1202) [reply src/views/PurchaseOrderDetail.vue]: Made the requested change

File: src/views/PurchaseOrderDetail.vue [hunk 2]
```diff
@@ -175,6 +176,12 @@ export default defineComponent({
     })
   },
   methods: {
+    getNumberOfItemsReceived(item: any){
```
Comment (adityasharma7) [inline src/views/PurchaseOrderDetail.vue:179]: Improve method name

File: src/views/PurchaseOrderDetail.vue [hunk 3]
```diff
@@ -175,6 +176,12 @@ export default defineComponent({
...
+      return (parseInt(item.quantityAccepted) +  this.getPOItemAccepted(item.productId))/(item.quantity)
+    },
+    areItemsReceivedGreaterThanItemsOrdered(item: any){
```
Comment (adityasharma7) [inline src/views/PurchaseOrderDetail.vue:182]: This method is not needed. We could use the above method only, when the fraction is greater than 1 we are over receiving

**Review by adityasharma7 — CHANGES_REQUESTED** [review]

**Review by adityasharma7 — APPROVED** [review]

**Pull request merged**